LiteCore/tests/QueryTest.cc

```diff
@@ -100,6 +100,25 @@ TEST_CASE_METHOD(DataFileTestFixture, "Query SELECT WHAT", "[Query]") {
     REQUIRE(num == 101);
 }
 
+TEST_CASE_METHOD(DataFileTestFixture, "Query SELECT All", "[Query]") {
+    addNumberedDocs(store);
+    Retained<Query> query{ store->compileQuery(json5("{WHAT: ['*', ['*', ['.num'], ['.num']]], WHERE: ['>', ['.num'], 10]}")) };
```

Comment:
The multiple asterisks here confused me at first! That highlights an ambiguity where `*` is the multiplication operator, except in the `WHAT` column list. That will prevent us from using the `*` property in expressions in the future. I think we should rename it.

`.*` would work well, since the prefix unambiguously identifies it as a property. What do you think?